source/text/Utf8.cpp

```diff
@@ -23,6 +23,14 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 using namespace std;
 
+
+
+namespace {
+	static constexpr char32_t BOM = 0x0000FEFF;
```

Comment:
static is not needed
```suggestion
	constexpr char32_t BOM = 0x0000FEFF;
```